src/App.jsx

```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+function getPrepearedGoods(goods, sortField, isReversed) {
+  const prepearedGoods = [...goods];
+
+  prepearedGoods.sort((a, b) => {
+    switch (sortField) {
+      case SORT_BY_ALPHABET:
+        return a.localeCompare(b);
+      case SORT_BY_LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPrepearedGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+
+  const makeSetSortField = (field) => () => setSortField(field);
+
+  const reverse = () => (
+    setIsReversed(prevIsReversed => !prevIsReversed)
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    const isResetButton = sortField !== '' || isReversed;
+
+    return isResetButton;
+  };
```

Comment:
```suggestion
    const isResetButton = sortField !== '' || isReversed;
```
enought just make variable. no need to make function and call in jsx